### PR TITLE
[Cosmos] disable api version check for new pylint requirements

### PR DESCRIPTION
**Pull request by simorenoh**
From https://azure.github.io/azure-sdk/python_design.html#python-client-constructor-api-version-argument-1

These guidelines will now begin to be enforced starting 04/15, so making this PR pre-emptively to unblock Cosmos once this check goes out (since it will block our pylint check in the pipelines). Can't merge this until those changes are out but wanted to get ahead of it.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/azure/cosmos/aio/cosmos_client.py
Comment (tjprescott) [inline sdk/cosmos/azure-cosmos/azure/cosmos/aio/cosmos_client.py:113]: Why not simply accept the api-version kwarg instead of disabling the check?

Comment (simorenoh) [reply sdk/cosmos/azure-cosmos/azure/cosmos/aio/cosmos_client.py:113]: We generally do not provide our users with the ability to choose which API to go against and would instead prefer to ignore this option, since allowing users to pass in this option could lead to more issues for us to debug for no real reason - going against any version that's not the current one is definitely not recommended

**Comment by simorenoh** [comment]
Date on this got pushed to next week

File: sdk/cosmos/azure-cosmos/azure/cosmos/aio/cosmos_client.py
Comment (johanste) [reply sdk/cosmos/azure-cosmos/azure/cosmos/aio/cosmos_client.py:113]: Is the latest API version guaranteed to be available across all cloud instances at the point where a new client library is released? And what backwards compat guarantees is the service making across api versions?

Comment (simorenoh) [reply sdk/cosmos/azure-cosmos/azure/cosmos/aio/cosmos_client.py:113]: @johanste not sure I understand what you're asking in regards to the SDK - we don't have control over the backend API or what changes they make. If newer versions affect the SDK and breaking changes need to be made, that's a separate issue than the scope of these changes - and only solidifies the fact we don't want users to have access to this setting.

That said, the current version that is to be used for the cosmos python SDK is guaranteed to be available (otherwise it wouldn't be the default).

Comment (tjprescott) [reply sdk/cosmos/azure-cosmos/azure/cosmos/aio/cosmos_client.py]: The standard (hence the guideline--hence the pylint rule) is that all Azure SDKs expose api_version as a parameter as a "best effort" option for customers who want to use the latest SDK with an older API version. Forcing customers to use an older version of the SDK to use an older API version would result in us having to hotfix a bunch of old SDKs for bugs whereas this model allows us to recommend that customers use the latest version of the SDK.

The pylint disable statements are there for false alarms. This is not a false alarm and thus it is hard to approve such a change, though I'll definitely defer to @annatisch and @johanste  on this one.

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/azure/cosmos/aio/cosmos_client.py]: @tjprescott @johanste  - This concept does not applies to Cosmos DB at all. Our SDK has more functionality than just serving as a wrapper for the REST calls - we don't support older API versions in our later SDKs. 

Also when we introduce direct mode for the SDK these changes would even become useless for instance as at that point there is no REST API call. 

This is not something that we do in any of our other SDKs.. (Java, .Net, NodeJS, GO SDK)

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/azure/cosmos/aio/cosmos_client.py]: @tjprescott @johanste  - This concept does not applies to Cosmos DB at all. Our SDK has more functionality than just serving as a wrapper for the REST calls - we don't support older API versions in our later SDKs. 

Also when we introduce direct mode for the SDK these changes would even become useless for instance as at that point there is no REST API call. 

This is not something that we do in any of our other SDKs.. (Java, .Net, NodeJS, GO SDK)

**Review by kushagraThapar — DISMISSED** [review]
This concept does not applies to Cosmos DB at all. Our SDK has more functionality than just serving as a wrapper for the REST calls - we don't support older API versions in our later SDKs.

Also when we introduce direct mode for the SDK these changes would even become useless for instance as at that point there is no REST API call.

This is not something that we do in any of our other SDKs.. (Java, .Net, NodeJS, GO SDK)

File: sdk/cosmos/azure-cosmos/azure/cosmos/http_constants.py
Comment (j82w) [reply sdk/cosmos/azure-cosmos/azure/cosmos/http_constants.py:275]: We should only expose the version the python SDK targets. Supporting any other version will likely break the SDK.

File: sdk/cosmos/azure-cosmos/azure/cosmos/aio/cosmos_client.py
Comment (ealsur) [reply sdk/cosmos/azure-cosmos/azure/cosmos/aio/cosmos_client.py]: Agree. Cosmos SDK due to their "thickness" contain logic that is compatible with the current service APIs released in prod. Before any new service API release, the team performs validations that the current existing SDKs will still function properly with any new service release. Some features in the current version in prod on the service might map to some SDK capability that if the customer decides to change the `api_version` would be broken. Breaking the SDK by setting a random lower `api_version` makes little sense to me.
Cosmos DB has no guidance anywhere that guides users to set X or Y `api_version` nor there is any need for such a guidance as far as I understand.

File: sdk/cosmos/azure-cosmos/azure/cosmos/http_constants.py
Comment (j82w) [inline sdk/cosmos/azure-cosmos/azure/cosmos/http_constants.py:280]: What is the point having these versions when the SDK does not support them?

Comment (simorenoh) [reply sdk/cosmos/azure-cosmos/azure/cosmos/http_constants.py:280]: @j82w Thinking from the point of view of someone who joined the SDK without much guidance - if in the future we want to see which api versions support different changes (specially for future versions, once the SDK is merge proof for instance), having the versions handy within the SDK seems like a good move (rather than going to the .NET SDK to check these).

The older versions I have no issues with removing, but the newer ones I would argue for keeping for the case I outlined before, and regardless we're not giving users a way to use these versions.

I guess the question would be if there is some public/ microsoft internal document where a new Cosmos developer would be able to find these, and if so then this would not be needed.

File: sdk/cosmos/azure-cosmos/azure/cosmos/http_constants.py
Comment (johanste) [inline sdk/cosmos/azure-cosmos/azure/cosmos/http_constants.py:269]: Is this a public class? If so we should not make breaking changes just to make the linter happy.

**Review by johanste — CHANGES_REQUESTED** [review]
Let's make sure we are not doing breaking changes to make the linter happy. If this is a change to something that hasn't already been released, then it is goodness...

File: sdk/cosmos/azure-cosmos/azure/cosmos/http_constants.py
Comment (simorenoh) [reply sdk/cosmos/azure-cosmos/azure/cosmos/http_constants.py]: Removed this to move forward, will document later on if needed.

**Review by tjprescott — APPROVED** [review]
Tracking issue for this topic here: https://github.com/Azure/azure-sdk-for-python/issues/24094

**Pull request merged**
